gst/nnstreamer/tensor_if/gsttensorif.c

```diff
@@ -1107,13 +1134,36 @@ static gboolean
 gst_tensor_if_check_condition (GstTensorIf * tensor_if, GstBuffer * buf,
     gboolean * result)
 {
-  tensor_if_data_s cv = {.type = _NNS_END,.data._uint8_t = 0 };
+  if (tensor_if->cv == TIFCV_CUSTOM) {
+    GstMemory *in_mem[NNS_TENSOR_SIZE_LIMIT];
+    GstMapInfo in_info[NNS_TENSOR_SIZE_LIMIT];
+    GstTensorMemory in_tensors[NNS_TENSOR_SIZE_LIMIT];
+    guint i, j;
+
+    for (i = 0; i < tensor_if->in_config.info.num_tensors; i++) {
+      in_mem[i] = gst_buffer_peek_memory (buf, i);
+      if (FALSE == gst_memory_map (in_mem[i], &in_info[i], GST_MAP_READ)) {
+        for (j = 0; j < i; j++)
+          gst_memory_unmap (in_mem[j], &in_info[j]);
+        GST_WARNING_OBJECT (tensor_if, "Cannot map input memory buffer(%d)\n",
+            i);
+        return GST_FLOW_ERROR;
+      }
+      in_tensors[i].data = in_info[i].data;
+      in_tensors[i].size = in_info[i].size;
+      gst_memory_unmap (in_mem[i], &in_info[i]);
+    }
 
-  if (!gst_tensor_if_calculate_cv (tensor_if, buf, &cv)) {
-    GST_ERROR_OBJECT (tensor_if, " failed to calculate compared value");
-    return FALSE;
+    return tensor_if->custom.func (&tensor_if->in_config.info, in_tensors,
```

Comment:
Could you add a test with user custom data? I think passing user-defined data is omitted here

Reply:
I added the case in  `TEST (tensor_if_custom, normal_0)`. Thanks.